deploy/charts/rook-ceph/templates/securityContextConstraints.yaml

```diff
@@ -0,0 +1,95 @@
+# scc for the Rook and Ceph daemons
+# for creating cluster in openshift
+{{- if .Values.openshift }}
+kind: SecurityContextConstraints
+apiVersion: security.openshift.io/v1
+metadata:
+  name: rook-ceph
+allowPrivilegedContainer: true
+allowHostDirVolumePlugin: true
+allowHostPID: false
+# set to true if running rook with host networking enabled
+allowHostNetwork: false
+# set to true if running rook with the provider as host
+allowHostPorts: false
+priority:
+allowedCapabilities: ["MKNOD"]
+allowHostIPC: true
+readOnlyRootFilesystem: false
+# drop all default privileges
+requiredDropCapabilities: ["All"]
+defaultAddCapabilities: []
+runAsUser:
+  type: RunAsAny
+seLinuxContext:
+  type: MustRunAs
+fsGroup:
+  type: MustRunAs
+supplementalGroups:
+  type: RunAsAny
+volumes:
+  - configMap
+  - downwardAPI
+  - emptyDir
+  - hostPath
+  - persistentVolumeClaim
+  - projected
+  - secret
+users:
+  # A user needs to be added for each rook service account.
+  # This assumes running in the default sample "rook-ceph" namespace.
+  # If other namespaces or service accounts are configured, they need to be updated here.
+  - system:serviceaccount:{{ .Release.Namespace }}:rook-ceph-system # serviceaccount:namespace:operator
+  - system:serviceaccount:{{ .Release.Namespace }}:default # serviceaccount:namespace:cluster
```

Comment:
Lines 43-46 require the cluster namespace if it is different from the operator namespace. So if they install the cluster in a different namespace than the operator, this won't work. I'm thinking we need an scc for the operator, and a separate scc to create with the cluster chart. Let's discuss.

Reply:
Updated